cli/targets/proto.js

```diff
@@ -184,8 +184,12 @@ function buildType(type) {
 }
 
 function buildField(field, passExtend) {
-    if (field.partOf || field.declaringField || field.extend !== undefined && !passExtend)
+    if (field.partOf && !field.partOf.isProto3Optional) {
```

Comment:
See this isProto3Optional? That's what's not preserved in the legacy required detection